### PR TITLE
ARC-36 add await to installation update

### PR DESCRIPTION
**Pull request by rachellerathbone**
This is a secondary (and hopefully final) PR to fix the flaky test in the app. Turns out, we weren't awaiting update in the install method so our super speedy tests were receiving the returned valued for the existing subscription instead of the renamed one.

### CODE REVIEW

**Comment by mboudreau** [comment]
@tebriel 

@rachellerathbone and I found the problem this morning while pairing.  Essentially it comes down to the tests being too fast and the `Installation.install` function not waiting for the subscriptions to be done updating before continuing.  Definitely a tricky one to solve :)

File: lib/models/installation.js [hunk 1]
```diff
@@ -95,11 +95,12 @@ class Installation extends Sequelize.Model {
...
       }).then(async (record) => {
         const subscriptions = await Subscription.getAllForClientKey(record.clientKey);
-        subscriptions.forEach(subscription => {
-          subscription.update({
+        for (const subscription of subscriptions) {
+          await subscription.update({
             jiraHost: record.jiraHost,
           });
-        });
+        }
```
Comment (mboudreau) [inline lib/models/installation.js:102]: I actually was thinking, if there's a lot of subscriptions, it would be slower than having everything in parallel since the loop would do everything sequentially.  An alternative would be to use `Promise.all` :)

```
await Promise.all(
            subscriptions.map(
                subscription => subscription.update({jiraHost: record.jiraHost})
            )
        );
```

**Review by mboudreau — CHANGES_REQUESTED** [review]
Small little change to do while we're there.  I also thought of looking at the `Subscription.install` function to see if it has the same issue, and it kinda does for the line `Subscription.findOrStartSync(subscription);` where it's missing an await before this.  Maybe a quick add? :)

File: lib/models/installation.js (hunk 1 above)
Comment (rachellerathbone) [reply lib/models/installation.js:102]: Good point. Let's not make that slower.

**Review by mboudreau — APPROVED** [review]
Awesome :D

File: test/unit/models/index.test.js [hunk 2]
```diff
@@ -72,6 +66,12 @@ describe('test installation model', () => {
     });
   });
 
+  afterEach(async () => {
+    // Clean up the database
+    await Installation.truncate({ cascade: true, restartIdentity: true });
+    await Subscription.truncate({ cascade: true, restartIdentity: true });
```
Comment (tebriel) [inline test/unit/models/index.test.js:72]: The only time I can see this being a problem is if the tests failed due to an exception and did not exit cleanly. This means the next run of the tests might have state from the previous run. Is there anything we can/should do about that?

File: lib/models/installation.js [hunk 3]
```diff
@@ -95,11 +95,8 @@ class Installation extends Sequelize.Model {
         jiraHost: payload.host,
       }).then(async (record) => {
         const subscriptions = await Subscription.getAllForClientKey(record.clientKey);
-        subscriptions.forEach(subscription => {
-          subscription.update({
-            jiraHost: record.jiraHost,
-          });
-        });
+        await Promise.all(subscriptions.map(subscription => subscription.update({ jiraHost: record.jiraHost })));
```
Comment (tebriel) [inline lib/models/installation.js:98]: Nice find, great improvement!

File: test/unit/models/index.test.js (hunk 2 above)
Comment (mboudreau) [reply test/unit/models/index.test.js:72]: Even if the test fails, the `afterEach` is still called.  It's better to do it this way as it's the recommended testing approach for all testing frameworks - start with clean slate, end with clean slate.  

The problem with having it in the `beforeEach` is that when another test suite starts it will have some state already there, where the assumption is that there would be none.  If another suite installed a models that this one doesn't know about, you'll have to truncate that one also, leading to adding extra code just for cleaning up another suite.

Comment (tebriel) [reply test/unit/models/index.test.js:72]: Yes, agreed on all counts, I meant in the case of a catastrophic exception (for example a ctrl-c). We probably don’t need to optimize for that case, but I wanted to raise the idea just in case.

**Review by tebriel — APPROVED** [review]

File: test/unit/models/index.test.js (hunk 2 above)
Comment (mboudreau) [reply test/unit/models/index.test.js:72]: Right.  We were going to address that later when rewriting the tests so that it doesn't have a dependency on an external postgres service for them to run :)
